include/jemalloc/internal/safety_check.h

```diff
@@ -0,0 +1,26 @@
+#ifndef JEMALLOC_INTERNAL_SAFETY_CHECK_H
+#define JEMALLOC_INTERNAL_SAFETY_CHECK_H
+
+void safety_check_fail(const char *format, ...);
+/* Can set to NULL for a default. */
+void safety_check_set_abort(void (*abort_fn)());
+
+JEMALLOC_ALWAYS_INLINE void
+safety_check_set_redzone(void *ptr, size_t usize, size_t bumped_usize) {
```

Comment:
I'm worried that this might increase the memory overhead from profiling. IIRC, 3 of the 4 pages may not be touched at all. Maybe let's only populate and check the first page?

Reply:
I missed that the red zone is only 32B. Never mind.